pdns/misc.cc

```diff
@@ -845,7 +845,6 @@ void addCMsgSrcAddr(struct msghdr* msgh, void* cmsgbuf, const ComboAddress* sour
     memset(pkt, 0, sizeof(*pkt));
     pkt->ipi6_addr = source->sin6.sin6_addr;
     pkt->ipi6_ifindex = itfIndex;
-    msgh->msg_controllen = cmsg->cmsg_len; // makes valgrind happy and is slightly better style
```

Comment:
The initial commit introducing this line, 9af87ad80c04f5e52f26335dd0d94eb454bbec90, points to an interesting discussion on the matter: https://codereview.chromium.org/3026044 
rfc3542 states, in section 20.2 "The cmsghdr Structure": "While sending an application may or may not include padding at the end of last ancillary data in msg_controllen and implementations must accept both as valid.", and then uses CMSG_SPACE() in 21.1.